core/Config.php

```diff
@@ -514,16 +514,25 @@ protected function getUncached($class, $name, $sourceOptions, &$result, $suppres
 			if ($extraSources) $sources = array_merge($sources, $extraSources);
 		}
 
-		$value = $nothing = null;
-
 		foreach ($sources as $staticSource) {
-			if (is_array($staticSource)) {
+//			if (is_array($staticSource)) {
+//				$value = isset($staticSource[$name]) ? $staticSource[$name] : $nothing;
+//			}
+//			else {
+//				foreach ($this->staticManifests as $i => $statics) {
+//					$value = $statics->get($staticSource, $name, $nothing);
+//					if ($value !== $nothing) break;
+//				}
+//			}
+
+			if(is_array($staticSource)) {
 				$value = isset($staticSource[$name]) ? $staticSource[$name] : $nothing;
-			}
-			else {
-				foreach ($this->staticManifests as $i => $statics) {
-					$value = $statics->get($staticSource, $name, $nothing);
-					if ($value !== $nothing) break;
+			} else if(class_exists($staticSource) && property_exists($staticSource, $name)) {
+				$property = new ReflectionProperty($staticSource, $name);
+				if($property->isStatic() && $property->isPrivate()) {
+					$property->setAccessible(true);
+					$value = $property->getValue();
+					$property->setAccessible(false);
```

Comment:
I'd have thought this was unnecessary as it doesn't actually change the visibility of the property for the class... 

citation: http://php.net/manual/en/reflectionproperty.setaccessible.php#110414